packages/vite/src/node/plugins/resolve.ts

```diff
@@ -124,21 +124,31 @@ export function resolvePlugin(baseOptions: InternalResolveOptions): Plugin {
         const fsPath = path.resolve(basedir, id)
         // handle browser field mapping for relative imports
 
-        if ((res = tryResolveBrowserMapping(fsPath, importer, options, true))) {
-          return res
-        }
+        const pathFromBasedir = normalizePath(fsPath).slice(basedir.length)
+        if (pathFromBasedir.startsWith('/node_modules/')) {
```

Comment:
Not sure if this will work in monorepo case?

Reply:
Do you have a failing repro after this PR? 
I think one possible improvement is to trigger the node resolution at this point, but only use it if the resolved path is the same one as the one being imported

Reply:
I pushed this change, it should avoid direct imports to be interpreted as packages from other node_modules folders.